### ISSUE TITLE
Fast nms implementation wrong

### ISSUE THREAD

**Issue by davidgill97** [ultralytics/ultralytics#22536]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

`TorchNMS.fast_nms` seems wrong when `use_triu=False`. Specifically, it computes sorted index for scores, but does not actually sort it, zeroing out wrong scores. https://github.com/ultralytics/ultralytics/blob/c0d1d27abebc6777c9ba51777f1eb33e47bd9940/ultralytics/utils/nms.py#L234 

To fix this bug, scores must be sorted in advance:
```python
scores, sorted_idx = torch.sort(scores, descending=True)
```

### Environment

N/A

### Minimal Reproducible Example

N/A

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @davidgill97, thanks for reporting this and for your interest in Ultralytics 🚀! This is an automated response to help you get support faster. An Ultralytics engineer will review and assist you here soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it. Since your report references `TorchNMS.fast_nms` with `use_triu=False`, please include:
- Exact ultralytics and PyTorch versions
- OS, CUDA/CUDNN, GPU/CPU details
- A short, self-contained snippet that reproduces the behavior
- Expected vs. actual results, including any logs or assertions
- Whether this persists after upgrading to the latest package (see below) 🙏

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- Notebooks with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- Google Cloud Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- Amazon Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- Docker Image. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for the detailed note and suggested approach 🙌. If you’re open to it, you can also check the “Are you willing to submit a PR?” box and include your minimal change with the MRE-backed verification—this helps us review and iterate faster.

**Comment by Y-T-G** [comment]
Duplicate of https://github.com/ultralytics/ultralytics/issues/22125

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has been merged in the pull request [ultralytics 8.3.223 — Sort scores in FastNMS with use_triu=False](https://github.com/ultralytics/ultralytics/pull/22537) by @Y-T-G with contributions from @glenn-jocher. 

Key changes that may resolve the issue:
- Corrected FastNMS indexing by operating on a sorted view of scores and writing results back to the original tensor to keep state consistent for NMSModel.
- Ensures topk selection aligns with the sorted indices, improving correctness of suppression and preventing incorrect detections.
- Maintains fixed-length index returns to avoid TFLite reshape errors, improving stability in TFLite and NMSModel workflows.
- Version bumped to 8.3.223.

How to test the fix:
- Upgrade to the latest release (once 8.3.223 is available on PyPI):
  - `pip install -U ultralytics>=8.3.223`
- Or install the latest code from main:
  - `pip install git+https://github.com/ultralytics/ultralytics.git@main`
- Alternatively, wait for the next official PyPI release if not comfortable using main.

Please share whether these changes resolve the issue in your environment. If problems persist, any logs, minimal repro steps, or environment details would be very helpful.

Thanks for reporting this and helping improve reliability across inference pipelines. 🙏
